### ISSUE TITLE
Mate panel does not autohide and even hide on hide button press in wayland, wayfife 0.8

### ISSUE THREAD

**Issue by mark-herbert42** [mate-desktop/mate-panel#1435]
On gentoo, wayfire 0.8.  Compiled with wayland support and somehow works.

**Comment by lukefromdc** [comment]
I did not do anything with the panel hide/autohide code in my wayland porting work, so this is just an example of a feature not yet implemented in wayland. If anyone else wants to help with this, PR's or even code snippets are most welcome

**Comment by mark-herbert42** [comment]
looks like xfce managed to do autohide in wayland - their new panel do it like a charm. They had a bad bug in previous version when panel was getting out of control after being sown up from hide and start blinking and stealing focus as mad - but with the current release it just works.

Native wayfire panel - wf-shell - hides also, but they have issue - if panel is hidden sometimes it get stuck there and it takes several moves to unhide it again. There is a hack for wf-panel that makes it remaining as 1 pixel line istead of complete hide - and with it the autohide works.  Anyway it uses C++ code so will be harder to port.

So it is possible to do under wayland, and ut does not require full rework of the code - at least xfce did not do a full rework. Me knowledge is still not enough to propose the PR - I am trying my best but nothing working so far. But at least with xfce experience it is clear that it is possible.

And I agree that hide buttons are not needed in wayland - as I see the way code made it is very hard to develop this hiding animations in specified locations, so the most important thing seems to be simple autohide of top/bottom panels. The main reason is the growing popularity of OLED latops, and panel is the most static element of the screen so it will shure be first to burn in.

**Comment by lukefromdc** [comment]
I myself don't know how to make a wayland app being rendered entirely offscreen detect the mouse cursor going to an edge. Only way I can think of would be to leave one or two lines of pixels showing, maybe turn everything transparent in those lines to protect the screen. Hiding icons that go right to the edge using transparency might not be easy.

How did XFCE do this?

**Comment by mark-herbert42** [comment]
https://gitlab.xfce.org/xfce/xfce4-panel/-/commit/1d9e13db5ee2a69ffa239490512d092b28805590

One of the commits with some comments.  But that panel does not have all those fancy modes as Xorg version of Mate panel does - it does not have those hide buttons and can not slide-hide to left of right - just the top panel hides to top, bottom - to bottom.

And yes you right about wayland rendering entirely offscreen - the story with wf-shell demonstrate that wayfire team could not make unhiding reliable so they still need workaround leaving 1 pixel line onscreen. This workaround did not land into wf-shell mainstream by the way, so vanilla wf-panel autohide mode is almost unusable. Well - maybe not completely unusable but certainly annoying. If you hide and immediately unhide - it works, but if it hides and you do some focus changes while wf-panel is hidden - it looses the ability to sense the mouse. So to unhide you need to do it couple times, click on somethig, wait for something - and finally it unhides. But thats definitely not the best way, 1 pixel line is a pretty fine trade-off as the modern OLEDs are mostly 3-4К resolution.

**Comment by lukefromdc** [comment]
Hide buttons might be the one thing that works,maybe make them almost transparent when the panel is hidden to minimize burn-in.

Note that at least on my setup there are a lot of static desktop icons and they are what I get the most visible burn-in from. Even if the
icons change I tend to keep them in the same positions.

**Comment by mark-herbert42** [comment]
thats right - but desktop icons is something that you can use or not use. I do not use icons on destop for pretty long time now, and switched off caja desktop paiting in favor of wayfire's wf-background and my wallpapers rotate every 5 mins.  Actually I 've loved that feature and used it a lot - and still use it on corporate Windows laptop - but on Linux I prefer to use multiple workspaces (currently 9) - and with this setup desktop icons seems to be less interesting approach.  On Xorg Mate I used brisk menu with a very convenient "Favorites" plane instead of desktop shortcuts.  Ufortunately this one is also not functional in wayland session - but that's not a native Mate component but separate app.

But it is kind of impossible to use Mate without a panel. Panel  is  a central user interface component , the backbone of the whole DE....

**Comment by lukefromdc** [comment]
Well, you can't prevent burn-in without an empty desktop with a background that itself won't burn in any contrasting items. That means no icons on the desktop AND a hidden panel AND a monocolor background. Note that I simply do not have time to handle all of this alone

**Comment by mark-herbert42** [comment]
Yes - all the developments Mate needs are beyond one person ability, no doubt of this.

But as for this point - no icons  and backroud is there. You have a control option for caja stop painting your desktop, and then caja is not stating all this story with backgroud things and icons if you do not want them.  So the only missing thing here is panel hiding.

But that's for sure not the only development Mate needs to be wayland-compatible as there are a lot of applets and apps that does not work at all in wayland and will not most likely as they need to be completely redesigned. Input control, monitor control, power manager, screen-saver/locker , screenshot, pack of applets....  Defnitely volume of work  not for 1 person who do it as a pet project, not a full time job.

**Comment by mark-herbert42** [comment]
checked with the latest xfce4-panel v4.21 - it also leaving couple of pexels visible - so seems thats the only way to make wayland panel hide-unhide reliable. If it is fully  hidden - then the only way to unhide is integrating the panel with compositor - making a kind of gnome-shell which does not look as good idea, we already have gnome for this. The strong part of Mate is that it can allow mixing components with other DEs and use of different compositors or WM's in Xorg.  I always used it with compiz in Xorg era - much better experience than with minimalistic Marco.  

Xfce panel works OK with wayfire - Xfce desktop itself now migrated to labwc as main compositor. But their auto-hide design works in any compositor I suppose, at least any wlroots one.
